lib/grpc_reflection/service/builder.ex

```diff
@@ -104,10 +120,75 @@ defmodule GrpcReflection.Service.Builder do
       root_symbols = %{symbol => response}
       root_files = %{(symbol <> ".proto") => response}
 
-      {%Agent{files: root_files, symbols: root_symbols}, referenced_types}
+      extension_file = symbol <> "Extension.proto"
+
+      {extension_numbers, extension_payload} =
+        process_extensions(mod, symbol, extension_file, descriptor)
+
+      {root_extensions, root_files} =
+        if extension_numbers != nil and extension_payload != nil do
+          {
+            %{symbol => extension_numbers},
+            Map.put(root_files, extension_file, %{file_descriptor_proto: [extension_payload]})
+          }
+        else
+          {%{}, root_files}
+        end
+
+      {%Agent{files: root_files, symbols: root_symbols, extensions: root_extensions},
+       referenced_types}
     end)
   end
 
+  defp process_extensions(
+         mod,
+         symbol,
+         extension_file,
+         %Google.Protobuf.DescriptorProto{extension_range: extension_range} = descriptor
+       )
+       when extension_range != [] do
+    unencoded_extension_payload = %Google.Protobuf.FileDescriptorProto{
+      name: extension_file,
+      package: package_from_name(symbol),
+      dependency: [symbol <> ".proto"],
+      syntax: get_syntax(mod)
+    }
+
+    {extension_numbers, extension_files} =
+      Enum.unzip(
+        for %Google.Protobuf.DescriptorProto.ExtensionRange{
+              start: start_index,
+              end: end_index
+            } <- descriptor.extension_range,
+            index <- start_index..end_index,
+            {_, ext} <- List.wrap(Protobuf.Extension.get_extension_props_by_tag(mod, index)) do
+          {index, convert_to_field_descriptor(symbol, ext)}
+        end
+      )
+
+    # type: 11 means message_type
```

Comment:
I think a `is_message_descriptor?/1` function may be a bit easier to read

maybe something like
```
for ext <- extension_files, is_message_descriptor?(ext), do
    ext.type_name
    |> convert_symbol_to_type_name()
    |> String.to_existing_atom()
    |> then(& &1.descriptor())
end
```
So we aren't embedding message dection logic in this function